### ISSUE TITLE
Consider convert underscore to dash in custom html attributes

### ISSUE THREAD

**Issue by yerlaser** [AndyObtiva/glimmer-dsl-xml#3]
Hi,

Would it be possible to convert underscore to dash in custom html attributes, like this:
div(some_attr: 'value') {'text'}
to
<div some-attr="value">text</div>

**Comment by AndyObtiva** [comment]
You can already do that like this:

```ruby
div('some-attr': 'value') {'text'}
```

Output:

```html
<div some-attr="value">text</div>
```

Or this:

```ruby
div('some-attr' => 'value') {'text'}
```

Output:

```html
<div some-attr="value">text</div>
```

Otherwise, I do not think it is a good idea to auto-convert underscores to dashes in attributes because I thought underscores could still be used in HTML/XML attributes sometimes (but not as first character in attribute name). I will do some more research to confirm.

Update: It seems that underscores are legal in attribute names after the first letter, so I would not want to auto-convert to dashes as that might be confusing to people who expect underscores to show up. Using quotes with : or => should work well enough to support dashes for now. I am closing this issue now, but if you still have more questions about it, please feel free to continue to respond while it is closed.

**Comment by AndyObtiva** [comment]
Note the update added to previous message. Closing.

**Comment by yerlaser** [comment]
OK. That's fair enough. I can do "sed" and convert them manually for now.
Thank you!

**Comment by yerlaser** [comment]
After reading a bit, though, it seems that:
1) using underscores in attributes names is discourage by many
2) using dashes as custom attribute separator is mandated

This leads me to believe that majority of people will want auto underscore conversion.
Those who do need to retain them, will most probably be a minority and can use the workaround with quotes that you mentioned above.

This is my thoughts. Maybe you can keep the issue open so that others can give their opinion.

**Comment by AndyObtiva** [comment]
Fair enough! 

After giving this some more thought, I think I can implement as a global glimmer config option, but not as an initial default.

I added to [TODO.md](https://github.com/AndyObtiva/glimmer-dsl-xml/blob/master/TODO.md) as "Support global glimmer config option to auto convert xml attribute underscores to dashes"

For example:

```ruby
require 'glimmer-dsl-xml'

Glimmer::Config.xml_attribute_underscore = '-'

include Glimmer

puts html { div(data_bind: 'name') {"Sean"} }
```

Output:

```
<div data-bind: "name">Sean</div>
```

**Comment by AndyObtiva** [comment]
Implemented in [v1.3.1](https://rubygems.org/gems/glimmer-dsl-xml/versions/1.3.1) and documented over here:

https://github.com/AndyObtiva/glimmer-dsl-xml#glimmer-config

Cheers.

**Comment by yerlaser** [comment]
Wow, that was quick!
Thanks a lot and have a good time of day!
